src/writer.rs

```diff
@@ -96,6 +138,14 @@ impl<W: Write> Writer<W> {
 
     #[inline]
     fn write_wrapped(&mut self, before: &[u8], value: &[u8], after: &[u8]) -> Result<usize> {
-        Ok(self.writer.write(before)? + self.writer.write(value)? + self.writer.write(after)?)
+        Ok(
+            if self.should_line_break {
+                self.writer.write(b"\n")? + self.writer.write(self.indents.as_slice())?
```

Comment:
Better to use `writeln!` macro to use the platform new line.

Reply:
According to `writeln!` doc, "`On all platforms, the newline is the LINE FEED character (\n/U+000A) alone (no additional CARRIAGE RETURN (\r/U+000D).`". According to https://stackoverflow.com/questions/33867076/is-there-a-format-spec-that-will-output-a-os-specific-newline/33867420#33867420 , most Windows applications should already handle `"\n"`.